source/rust_verify/tests/arch_word_bits.rs

```diff
@@ -0,0 +1,203 @@
+
+            assert(i8::MIN == -0x80);
+            assert(i16::MIN == -0x8000);
+            assert(i32::MIN == -0x80000000);
+            assert(i64::MIN == -0x8000000000000000);
+            assert(i128::MIN == -0x80000000000000000000000000000000);
+            assert(isize::MIN == -0x80000000 || isize::MIN == -0x8000000000000000);
+
+            assert(u8::MAX == 0xff);
+            assert(u16::MAX == 0xffff);
+            assert(u32::MAX == 0xffffffff);
+            assert(u64::MAX == 0xffffffffffffffff);
+            assert(u128::MAX == 0xffffffffffffffffffffffffffffffff);
+            assert(usize::MAX == 0xffffffff || usize::MAX == 0xffffffffffffffff);
+
+            assert(i8::MAX == 0x7f);
+            assert(i16::MAX == 0x7fff);
+            assert(i32::MAX == 0x7fffffff);
+            assert(i64::MAX == 0x7fffffffffffffff);
+            assert(i128::MAX == 0x7fffffffffffffffffffffffffffffff);
+            assert(isize::MAX == 0x7fffffff || isize::MAX == 0x7fffffffffffffff);
+
+            assert(u8::BITS == 8);
+            assert(u16::BITS == 16);
+            assert(u32::BITS == 32);
+            assert(u64::BITS == 64);
+            assert(u128::BITS == 128);
+            assert(usize::BITS == 32 || usize::BITS == 64);
+
+            assert(i8::BITS == 8);
+            assert(i16::BITS == 16);
+            assert(i32::BITS == 32);
+            assert(i64::BITS == 64);
+            assert(i128::BITS == 128);
+            assert(isize::BITS == 32 || isize::BITS == 64);
+
+            // Constant should work in exec-mode
+            let x = isize::BITS;
+            assert(x == 32 || x == 64);
+        }
+
+        fn arch_fail_1() {
+            assert(arch_word_bits() == 32); // FAILS
+        }
+        fn arch_fail_2() {
+            assert(arch_word_bits() == 64); // FAILS
+        }
+        fn arch_fail_3() {
+            assert(arch_word_bits() == 32) by(compute); // FAILS
+        }
+        fn arch_fail_4() {
+            assert(arch_word_bits() == 64) by(compute); // FAILS
+        }
+        fn arch_fail_5() {
+            assert(isize::MIN == -0x8000000000000000); // FAILS
+        }
+    } => Err(err) => assert_fails(err, 5)
+}
+
+test_verify_one_file! {
+    #[test] test_set_to_32 verus_code! {
+
+        fn test1() {  // ARCH-WORD-BITS-32
+            assert(arch_word_bits() == 32);
+        }
+        fn test2() {
+            assert(arch_word_bits() == 32) by(compute);
+        }
+
+        fn test_constants() {
+            assert(isize::MIN == -0x80000000);
+            assert(usize::MAX == 0xffffffff);
+            assert(isize::MAX == 0x7fffffff);
+            assert(usize::BITS == 32);
+            assert(isize::BITS == 32);
+        }
+
+        fn fail1() {
+            assert(arch_word_bits() == 64); // FAILS
+        }
+    } => Err(err) => assert_fails(err, 1)
+}
+
+test_verify_one_file! {
+    #[test] test_set_to_64 verus_code! {
+
+        fn test1() {  // ARCH-WORD-BITS-64
```

Comment:
Minor: can you move the comment one line up to make it clear that it applies to the whole test? And can you add a comment stating it's a load-bearing comment (i.e. it affects the test environment)?

Reply:
i don't know why but it doesn't work if it's on its own line.

It's definitely jank though, I was thinking of just making a better way to pass in arguments in the test_verify_one_file! macro instead.